src/services/spheres.js

```diff
@@ -178,8 +178,21 @@ class Spheres {
       detailedLocation,
     );
 
+    let chartForIsland;
+    if (!_.isNil(itemAtLocation)
+        && LogicHelper.isRandomizedCharts()
+        && itemAtLocation.match(/(Treasure|Triforce) Chart (\d)+/)) {
+      const island = this.state.getIslandForChart(itemAtLocation);
+      if (!_.isNil(island)) {
+        chartForIsland = `Chart for ${island}`;
```

Comment:
The logic to convert an island name to `Chart for ${island}` should be added as a method to `LogicHelper`.